SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -3182,12 +3263,94 @@ FUnrealObjectRef USpatialNetDriver::GetCurrentPlayerControllerRef()
 
 void USpatialNetDriver::PushCrossServerRPCSender(AActor* SenderActor)
 {
-	GSenderStack.Add(SenderActor);
+	using namespace SpatialNetDriverPrivate;
+	check(!GSenderActor.IsSet());
+	GSenderActor.Emplace(SenderActorDesc(SenderActor, SenderActorDesc::Sender));
+}
+
+void USpatialNetDriver::PopCrossServerRPCSender()
+{
+	using namespace SpatialNetDriverPrivate;
+	GSenderActor.Reset();
+}
+
+void USpatialNetDriver::PushDependentActor(AActor* Dependent)
+{
+	using namespace SpatialNetDriverPrivate;
+	check(!GSenderActor.IsSet());
+	GSenderActor.Emplace(SenderActorDesc(Dependent, SenderActorDesc::Dependent));
+}
+
+void USpatialNetDriver::PopDependentActor()
+{
+	using namespace SpatialNetDriverPrivate;
+	GSenderActor.Reset();
+}
+
+bool USpatialNetDriver::NeedWriteFence(AActor* Actor, UFunction* Function)
```

Comment:
Looking at this function name i'd expect it to just check if the function param included NetWriteFence, but this seems to be checking if in this specific instance if we need to implement the write fence. What about ShouldImplementWriteFence?

Reply:
So function flag checking is done in GetFunctionCallspace, I have been trying to keep it there.
"Implement" sounds weird to me in this context. How does RPCCallNeedWriteFence sounds ?

Reply:
Yep sounds good